src/main/java/com/eternalcode/parcellockers/parcel/ParcelRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,154 @@
+package com.eternalcode.parcellockers.parcel;
+
+import com.eternalcode.parcellockers.database.JdbcConnectionProvider;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ParcelRepositoryJdbcImpl implements ParcelRepository {
+
+    private final JdbcConnectionProvider jdbcConnectionProvider;
+
+    private ParcelRepositoryJdbcImpl(JdbcConnectionProvider jdbcConnectionProvider) {
+        this.jdbcConnectionProvider = jdbcConnectionProvider;
+    }
+
+    @Override
+    public CompletableFuture<Void> save(Parcel parcel) {
+        return CompletableFuture.runAsync(() -> {
+            ParcelMeta meta = parcel.getMeta();
+            this.jdbcConnectionProvider.executeUpdate("INSERT INTO `parcels` (`uuid`, `name`, `description`, `priority`, `receiver`, `size`, `entryLocker`, `destinationLocker`, `sender`) VALUES (%u, %n, %d, %p, %r, %s, %e, %dst, %sn)"
```

Comment:
Tutaj powinien być zrobiony preparedstatement a nie taki brzydki string

Reply:
metoda executeUpdate() wykonuje PreparedStatement

Reply:
Chodzi o prepared statement z setowanymi wartościami ps.setString itp a nie to coś co wysyłasz w jednej linijce i robisz milion replace'ów

Reply:
Czyli będzie trzeba to robić ręcznie, bez pomocy JdbcConnectionProvidera?